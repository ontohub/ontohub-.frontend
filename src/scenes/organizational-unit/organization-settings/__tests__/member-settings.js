import { PureMemberSettings, MemberSettings } from "../member-settings";

describe("PureMemberSettings", () => {
  set("component", () => (
    <Router>
      <PureMemberSettings
        addOrganizationMember={addOrganizationMember}
        handleChange={handleChange}
        handleSubmit={handleSubmit}
        me={me}
        memberships={memberships}
        popup={popup}
        removeOrganizationMember={removeOrganizationMember}
        setPopup={setPopup}
        values={values}
      />
    </Router>
  ));

  set("renderedComponent", () => render(component));
  set("wrapper", () => mount(component));

  set("addOrganizationMember", () => jest.fn());
  set("handleChange", () => jest.fn());
  set("handleSubmit", () => jest.fn());
  set("me", () => Factories.User());
  set("memberships", () => [
    {
      member: Factories.User({ id: "ada", displayName: "Ada Lovelace" }),
      role: "admin"
    }
  ]);
  set("popup", () => null);
  set("removeOrganizationMember", () => jest.fn());
  set("removeOrganizationMemberWrapper", () =>
    jest.fn(() => removeOrganizationMember)
  );
  set("setPopup", () => jest.fn());
  set("values", () => ({ member: "", role: "read" }));

  it("matches the snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  describe("New member form", () => {
    set("newMemberWrapper", () => wrapper.find("#new_member"));

    it("Calls onChange when changing the role", () => {
      newMemberWrapper.find("Dropdown").simulate("change", {});
      expect(handleChange.mock.calls.length).toEqual(1);
    });

    it("Calls onChange when changing the name", () => {
      newMemberWrapper.find("input[name='member']").simulate("change", {});
      expect(handleChange.mock.calls.length).toEqual(1);
    });

    it("calls the onSubmit callback when the add button is clicked", () => {
      newMemberWrapper.find("Button").simulate("click", {});
      expect(handleSubmit.mock.calls.length).toEqual(1);
    });
  });

  describe("member list", () => {
    it("calls the addOrganizationMember callback when changing a role", () => {
      wrapper
        .find("Dropdown")
        .at(1)
        .simulate("change", {});
      expect(addOrganizationMember.mock.calls.length).toEqual(1);
    });

    describe("Confirmation popup", () => {
      it("opens the popup when removing a user", () => {
        wrapper
          .find("Button")
          .at(1)
          .simulate("click", {});
        expect(setPopup.mock.calls.length).toEqual(1);
      });

      describe("Open popup", () => {
        set("popup", () => "ada");

        it("calls remove member callbacks", () => {
          wrapper
            .find("Confirm")
            .instance()
            .props.onConfirm();

          expect(removeOrganizationMember.mock.calls.length).toEqual(1);
          expect(setPopup.mock.calls.length).toEqual(1);
        });

        it("closes the popup when clicking cancel", () => {
          wrapper
            .find("Confirm")
            .instance()
            .props.onCancel();

          expect(setPopup.mock.calls.length).toEqual(1);
        });
      });
    });
  });
});
