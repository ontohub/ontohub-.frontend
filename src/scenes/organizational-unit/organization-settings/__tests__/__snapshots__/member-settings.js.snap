// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PureMemberSettings matches the snapshot 1`] = `
.c0 {
  height: 0 !important;
  width: 0 !important;
  padding: 0 1px;
  margin-top: -1em !important;
}

<form>
  <h3
    class="ui header"
  >
    Organization Members
  </h3>
  <table
    class="ui compact table"
  >
    <thead
      class=""
    >
      <tr
        class=""
      >
        <th
          class=""
        >
          User
        </th>
        <th
          class="four wide"
        >
          Role
        </th>
        <th
          class="one wide"
        >
          
        </th>
      </tr>
    </thead>
    <tbody
      class=""
    >
      <tr
        class=""
        id="new_member"
      >
        <td
          class=""
        >
          <div
            class="ui small fluid input"
          >
            <input
              name="member"
              placeholder="Username"
              type="text"
              value=""
            >
              
            </input>
          </div>
        </td>
        <td
          class=""
        >
          <div
            aria-expanded="false"
            class="ui basic fluid selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-live="polite"
              class="text"
              role="alert"
            >
              Read
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            >
              
            </i>
            <div
              class="menu transition"
            >
              <div
                aria-checked="true"
                aria-selected="true"
                class="active selected item"
                role="option"
                style="pointer-events:all"
              >
                <span
                  class="text"
                >
                  Read
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events:all"
              >
                <span
                  class="text"
                >
                  Write
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events:all"
              >
                <span
                  class="text"
                >
                  Admin
                </span>
              </div>
            </div>
          </div>
        </td>
        <td
          class=""
        >
          <button
            class="ui primary button"
            role="button"
          >
            <i
              aria-hidden="true"
              class="plus icon c0"
            >
              
            </i>
          </button>
        </td>
      </tr>
      <tr
        class=""
      >
        <td
          class=""
        >
          <h4
            class="ui image header"
          >
            <img
              alt="Gravatar for undefined"
              class="react-gravatar image"
              height="50"
              src="//www.gravatar.com/avatar/3e3417d7ef77d5932a6734b916515ed5?d=retro&r=g&s=50"
              srcset="//www.gravatar.com/avatar/3e3417d7ef77d5932a6734b916515ed5?d=retro&r=g&s=100 2x"
              style="border-radius:2px"
              width="50"
            >
              
            </img>
            <div
              class="content"
            >
              <a
                aria-current="false"
                href="/ada"
              >
                Ada Lovelace
              </a>
              <div
                class="sub header"
              >
                ada
              </div>
            </div>
          </h4>
        </td>
        <td
          class=""
        >
          <div
            aria-expanded="false"
            class="ui basic fluid selection dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-live="polite"
              class="text"
              role="alert"
            >
              Admin
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            >
              
            </i>
            <div
              class="menu transition"
            >
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events:all"
              >
                <span
                  class="text"
                >
                  Read
                </span>
              </div>
              <div
                aria-checked="false"
                aria-selected="false"
                class="item"
                role="option"
                style="pointer-events:all"
              >
                <span
                  class="text"
                >
                  Write
                </span>
              </div>
              <div
                aria-checked="true"
                aria-selected="true"
                class="active selected item"
                role="option"
                style="pointer-events:all"
              >
                <span
                  class="text"
                >
                  Admin
                </span>
              </div>
            </div>
          </div>
        </td>
        <td
          class=""
        >
          <button
            class="ui negative button"
            role="button"
          >
            <i
              aria-hidden="true"
              class="trash icon c0"
            >
              
            </i>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</form>
`;
